docs/HACKING.md

```diff
@@ -143,3 +143,86 @@ $ make vmoverlay SKIP_INSTALL=1
 
 Of course, you can use this pattern for not just ostree but whatever else you'd
 like to install into the VM (e.g. bubblewrap, libsolv, etc...).
+
+## Using GDB with the rpm-ostree daemon
+
+If you're new to rpm-ostree, before using GDB, it may be helpful to review the
+[daemon architecture doc](../architecture-daemon.md) for an architecture recap.
+
+### Server-side (composes)
+
+Server-side composes do not use the daemon architecture and so one can naturally
+do e.g. `gdb --args rpm-ostree compose tree ...`. If using coreos-assembler, you
+can set the `COSA_RPMOSTREE_GDB` environment variable like this:
+
+```sh
+$ COSA_RPMOSTREE_GDB="gdb --args" cosa build
+```
+
+When cosa gets to the point of invoking rpm-ostree for the compose, it will call
+GDB instead.
+
+### Client-side
+
+On the client side, you need to use the `make vmsync` flow before using GDB
+because it also copies over the source files into `/root/sync`.
+
+For throwaway/fresh VMs, a simple approach is to just layer it using e.g.
```

Comment:
Absolutely easier but...to achieve our goals we need to make containers even easier than this.

Reply:
Yeah, hesitated putting this in. I can drop it if you strongly prefer or maybe we can just de-emphasize it a bit more. Probably still better to leave it in just for folks to be aware of their options. If the container stack is broken or you're new to all this, in a debugging situation it's nice to be able to just punt on all that and focus on your bug.

Reply:
Oh fine as is to keep!